components/WeekdayFromDateString.js

```diff
@@ -0,0 +1,27 @@
+import React, { useState, useEffect } from "react";
+
+export default function WeekdayFromDateString({ dateString }) {
+  const [weekday, setWeekday] = useState("");
+
+  useEffect(() => {
+    const dateTimeParts = dateString.split(" ");
+    const datePart = dateTimeParts[0];
+    const dateObject = new Date(datePart);
+
+    const weekdays = [
+      "Sonday",
+      "Monday",
+      "Tuesday",
+      "Wednesday",
+      "Thursday",
+      "Friday",
+      "Saturday",
+    ];
+    const weekdayIndex = dateObject.getDay();
+    const weekday = weekdays[weekdayIndex];
+
+    setWeekday(weekday);
+  }, [dateString]);
+
+  return <p>{weekday}</p>;
+}
```

Comment:
you don't need a state nor an effect here. Simply perform the calculations from line 7 till 21 directly in the component:
```js
const dateTimeParts = dateString.split(" ");
const datePart = dateTimeParts[0];
const dateObject = new Date(datePart);

const weekdays = [
  "Sonday",
  "Monday",
  "Tuesday",
  "Wednesday",
  "Thursday",
  "Friday",
  "Saturday",
];
const weekdayIndex = dateObject.getDay();
const weekday = weekdays[weekdayIndex];
return <p>{weekday}</p>;
}
```